src/descriptor/tr.rs

```diff
@@ -553,8 +553,10 @@ impl<'a, Pk: ToPublicKey, Ext: ParseableExt> TapLeafScript<'a, Pk, Ext> {
             TapLeafScript::Simplicity(sim) => {
                 let satisfier = crate::simplicity::SatisfierWrapper::new(satisfier);
                 let program = sim.satisfy(&satisfier).map_err(|_| Error::CouldNotSatisfy)?;
-                let program_and_witness_bytes = program.encode_to_vec();
-                Ok(vec![program_and_witness_bytes])
+                let (program_bytes,witness_bytes) = program.encode_to_vec();
+                let mut both = program_bytes;
+                both.extend(witness_bytes);
+                Ok(vec![both])
```

Comment:
I am not sure about this

Reply:
Oh, thanks for drawing attention to this. It does look wrong. I think the right thing is

```rust
let (program_bytes, witness_bytes) = program.encode_to_vec();
Ok(vec![program_bytes, witness_bytes])
```

Your code is equivalent to the old behavior, but we actually changed the consensus serialization so that now the program and witness appear in separate vector elements.

Reply:
changed and squashed in latest commit